File: OLiveDMC/ClientApp/src/app/providers/LocationService/location.service.ts
import { Injectable } from '@angular/core';
import { DataService } from '../dataservice/data.service';
import { ResponseModel } from 'src/app/model/ResponseModel';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class LocationService {


  constructor(private dataService: DataService) { }

  // -----------------------------------------------Country Methods--------------------------------------------------

  AddUpdateCountry(data){
    return <Observable<ResponseModel>> this.dataService.postFormData('Location/AddUpdateCountry', data);
  }


  GetAllCountry()
  {
    return <Observable<ResponseModel>> this.dataService.getData('Location/GetAllCountry');
  }
 
  deleteCountry(CountryId)
  {
    return <Observable<ResponseModel>> this.dataService.getData('Location/deleteCountry?Id='+CountryId);
  }

  GetCountryDetailByCountryId(CountryId)
  {
    return <Observable<ResponseModel>> this.dataService.getData('Location/GetCountryDetailByCountryId?Id='+CountryId);
  }


  // -----------------------------------------------State Methods--------------------------------------------------
  

  AddUpdateState(data){
    return <Observable<ResponseModel>> this.dataService.postFormData('Location/AddUpdateState', data);
  }


  GetAllState()
  {
    return <Observable<ResponseModel>> this.dataService.getData('Location/GetAllState');
  }
 
  deleteState(StateId)
  {
    return <Observable<ResponseModel>> this.dataService.getData('Location/deleteState?Id='+StateId);
  }

  GetStateDetailByStateId(StateId)
  {
    return <Observable<ResponseModel>> this.dataService.getData('Location/GetStateDetailByStateId?Id='+StateId);
  }


  // -----------------------------------------------City Methods--------------------------------------------------

  AddUpdateCity(data){
    return <Observable<ResponseModel>> this.dataService.postFormData('Location/AddUpdateCity', data);
  }


  GetAllCity()
  {
    return <Observable<ResponseModel>> this.dataService.getData('Location/GetAllCity');
  }
 
  deleteCity(CityId)
  {
    return <Observable<ResponseModel>> this.dataService.getData('Location/deleteCity?Id='+CityId);
  }

  GetCityDetailByCityId(CityId)
  {
    return <Observable<ResponseModel>> this.dataService.getData('Location/GetCityDetailByCityId?Id='+CityId);
  }
 
  GetStateByCountryId(countryId)
  {
    return <Observable<ResponseModel>> this.dataService.getData('Location/GetStateByCountryId?countryId='+countryId);
  }
  
  GetCityByStateId(stateId)
  {
    return <Observable<ResponseModel>> this.dataService.getData('Location/GetCityByStateId?stateId='+stateId);
  }


}

File: BusinessServices/Services/LocationService.cs
using BusinessDataModel.DB;
using BusinessDataModel.ViewModel;
using BusinessRespository.Infrastructure;
using BusinessServices.IServices;
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Text;

namespace BusinessServices.Services
{
    public class LocationService: ILocationService
    {
        private IUnitOfWork _unitOfWork;
        public IHttpContextAccessor __httpContextaccessor;
        private ResponseModel response;

        public LocationService(IUnitOfWork unitOfWork, IHttpContextAccessor httpContextAccessor)
        {
            _unitOfWork = unitOfWork;
            response = new ResponseModel();
            __httpContextaccessor = httpContextAccessor;
            //if (_httpContextaccessor.HttpContext.Session.GetString("Userdetail") != null)
            //    //{
            //    //    SessionUser = JsonConvert.DeserializeObject<Registration>(_httpContextaccessor.HttpContext.Session.GetString("Userdetail"));
            //    //}
        }

        //----------------------------------Country-----------------------------------------------------------------

        public ResponseModel AddUpdateCountry(Country obj)
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.AddUpdateCountry(obj);

                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }
        public ResponseModel GetAllCountry()
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.GetAllCountry();

                response.data = result.data;
                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }

        public ResponseModel deleteCountry(int? Id)
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.deleteCountry(Id);

                response.data = result.data;
                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }


        public ResponseModel GetCountryDetailByCountryId(int? Id)
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.GetCountryDetailByCountryId(Id);

                response.data = result.data;
                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }

        //----------------------------------State-----------------------------------------------------------------

        public ResponseModel AddUpdateState(State obj)
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.AddUpdateState(obj);

                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }
        public ResponseModel GetAllState()
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.GetAllState();

                response.data = result.data;
                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }

        public ResponseModel deleteState(int? Id)
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.deleteState(Id);

                response.data = result.data;
                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }


        public ResponseModel GetStateDetailByStateId(int? Id)
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.GetStateDetailByStateId(Id);

                response.data = result.data;
                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }


        //-----------------------------------------City-------------------------------------------------------------


        public ResponseModel AddUpdateCity(City obj)
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.AddUpdateCity(obj);

                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }
        public ResponseModel GetAllCity()
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.GetAllCity();

                response.data = result.data;
                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }

        public ResponseModel deleteCity(int? Id)
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.deleteCity(Id);

                response.data = result.data;
                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }


        public ResponseModel GetCityDetailByCityId(int? Id)
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.GetCityDetailByCityId(Id);

                response.data = result.data;
                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }
        
        public ResponseModel GetStateByCountryId(int? countryId)
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.GetStateByCountryId(countryId);

                response.data = result.data;
                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }

        public ResponseModel GetCityByStateId(int? stateId)
        {
            try
            {
                ResponseModel result = new ResponseModel();
                result = _unitOfWork.LocationRepository.GetCityByStateId(stateId);

                response.data = result.data;
                response.status = result.status;
                response.message = result.message;

            }
            catch (Exception ex)
            {
                response.message = ex.Message;
                response.status = Status.Error;
            }
            return response;
        }

    }
}
